### PR TITLE
[Modules] Update Network LocalNetworkGateway to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Update Network LocalNetworkGateway to new dependency approach

| Pipeline |
| - |
| [Network: LocalNetworkGateways](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.localnetworkgateways.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Network/localNetworkGateways/.test/common/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,60 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes')
```
Comment (MrMCake) [inline modules/Microsoft.Network/localNetworkGateways/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes')
```

File: modules/Microsoft.Network/localNetworkGateways/.test/min/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,49 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes')
```
Comment (MrMCake) [inline modules/Microsoft.Network/localNetworkGateways/.test/min/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes')
```

File: modules/Microsoft.Network/localNetworkGateways/.test/min/dependencies.bicep [hunk 3]
```diff
@@ -0,0 +1,14 @@
+@description('Optional. The location to deploy to.')
```
Comment (eriqua) [inline modules/Microsoft.Network/localNetworkGateways/.test/min/dependencies.bicep:1]: Please remove file, not needed for min test.

File: modules/Microsoft.Network/localNetworkGateways/.test/min/deploy.test.bicep [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+// =========== //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module resourceGroupResources 'dependencies.bicep' = {
```
Comment (eriqua) [inline modules/Microsoft.Network/localNetworkGateways/.test/min/deploy.test.bicep:27]: Please remove module, not used and not needed for min test.

Comment (MrMCake) [reply modules/Microsoft.Network/localNetworkGateways/.test/min/deploy.test.bicep:27]: done

File: modules/Microsoft.Network/localNetworkGateways/.test/min/dependencies.bicep (hunk 3 above)
Comment (MrMCake) [reply modules/Microsoft.Network/localNetworkGateways/.test/min/dependencies.bicep]: done

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**